### PR TITLE
[skrifa] tthint tests batch 2

### PR DESCRIPTION
**Pull request by dfrg**
More work on #803

Tests for move_point(), move_zp2_point(), move_original(), point_displacement(), SHP, SHC, SHZ, SHPIX, MSIRP, MDAP

### CODE REVIEW

File: skrifa/src/outline/glyf/hint/engine/mod.rs [hunk 1]
```diff
@@ -190,7 +190,7 @@ mod mock {
                 unscaled: vec![Default::default(); 32],
                 points: vec![Default::default(); 64],
                 point_flags: vec![Default::default(); 32],
-                contours: vec![32],
+                contours: vec![31],
```
Comment (cmyr) [inline skrifa/src/outline/glyf/hint/engine/mod.rs:193]: did somebody's tests uncover a bug?

**Review by cmyr — APPROVED** [review]
lgtm

File: skrifa/src/outline/glyf/hint/engine/mod.rs (hunk 1 above)
Comment (dfrg) [reply skrifa/src/outline/glyf/hint/engine/mod.rs:193]: A bug in the test support code but that counts for something?

Comment (cmyr) [reply skrifa/src/outline/glyf/hint/engine/mod.rs:193]: half points

**Pull request merged**